src/lib/support/Pool.h

```diff
@@ -0,0 +1,128 @@
+/*
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Defines a header only BitMapObjectPool.
+ */
+
+#pragma once
+
+#include <array>
+#include <assert.h>
+#include <atomic>
+#include <new>
+#include <stddef.h>
+
+namespace chip {
+
+template <class T, size_t N>
+class BitMapObjectPool
+{
+public:
+    BitMapObjectPool()
+    {
+        for (size_t word = 0; word * kAtomicSize < N; ++word)
```

Comment:
Is there a reason not to do:
```
for (auto & usage : mUsage) {
  usage.store(0);
}
```
and not mess around with the indices and making sure the indexing matches?

Reply:
I still need `word` as integer to check if it is the last member of `mUsage`

Reply:
Why do you need to check that in the constructor?  There is no different treatement of the last word in mUsage in the constructor; it's just setting all the words in mUsage to 0.